src/integer_mod_q/mat_zq/inverse.rs

```diff
@@ -11,11 +11,65 @@
 use super::MatZq;
 use crate::{
     integer::{MatZ, Z},
-    traits::{Concatenate, GetNumRows},
+    traits::{Concatenate, Gcd, GetNumRows},
 };
 use flint_sys::fmpz_mod_mat::fmpz_mod_mat_rref;
 
 impl MatZq {
+    /// Returns the inverse of the matrix if it exists (is square and
+    /// has a determinant co-prime to the modulus) and `None` otherwise.
```

Comment:
Is this a required property, I have not found it anywhere yet.
How does this behave for invertible matrices with negative determinant?, e.g. [[0, 1],[1, 0]]?

Reply:
Yes, since AB = I ⟹ det(A) det(B) = 1 mod p. Negative values are treated like non negative values